### ISSUE TITLE
Remove attestation specification from spec

### ISSUE THREAD

**Issue by leshi** [w3c/webauthn#108]
There seems to be growing consensus that detailed specification of attestation formats does not belong in this spec.

Perhaps one way is to say that all attestation statements have a `type` and a `value`.  Where type is an enum and `value` is a byte array.  Each `type` enum will specify an outside reference for how servers interpret `value`.  For example, for `type=android-safetynet`, `value` should be interpreted as per http://developer.android.com/training/safetynet/index.html.  The webauthn spec would give this mapping and that's it.

**Comment by rlin1** [comment]
I think we have discussed that and concluded to keep the secion in the spec.
Should we close this issue as "won't fix" then?

**Comment by vijaybh** [comment]
This came up again in an in-person discussion with Giri today. Giri made the excellent point that the current discussion around Android N attestation shows that this needs to remain extensible in future, so we may be well served by putting this in the registry as well. In that case we could also move out some (all?) of the attestation formats from the doc since this doc is not the primary reference for them. Let's discuss at the next call.

**Comment by vijaybh** [comment]
This was discussed near the end of the call today: http://www.w3.org/2016/07/27-webauthn-minutes.html

I will be sending out a PR for #126 and #127 which makes the attestation formats more modular from a spec perspective. Separately, we can use the list to discuss the question raised here about the best home for those modular definitions.

**Comment by equalsJeffH** [comment]
regarding https://github.com/w3c/webauthn/issues/108#issuecomment-233847032 ...
A registry will not be "a primary reference" for the specification of attestation formats -- and thus whether or not a IANA registry of webauthn attestation types exists is **orthogonal** to whether or not the actual specification of (some of) webauthn attestation formats remains in the webauthn spec. See also #155 and..
> wrt registries (was: Spec status)
> https://lists.w3.org/Archives/Public/public-webauthn/2016Jul/0308.html

I.e., if we keep the specification of the presently-known webauthn attestation formats in the webauthn spec, and then yet another attestation format is defined at some point in the future, **we *WILL NOT* have to update the webauthn spec**.

Rather, that new attestation format can be specified in a separate spec which also registers the new attestation format type in the IANA webauthn attestation types registry.  Thus the WebAuthn spec itself would not need to be updated.

**Comment by gmandyam** [comment]
@equalsJeffH 

Re:  "if we keep the specification of the presently-known webauthn attestation formats in the webauthn spec"

In my opinion, the only attestation format specification format that might be kept in the spec is packed, because it is defined in the WebAuthn standard itself.  Since the other types of attestations are defined by other organizations, the attestation registry can hold those references (to specifications for Android platform, TCG/TPM, etc).  

One possibility is to create a new W3C specification for packed attestation, and include the reference to it in the registry.  This way, we can be internally consistent with the registry approach.

**Comment by selfissued** [comment]
There's a reason that the contributed FIDO specs contained descriptions of the attestation formats that they did - because they're practically needed by implementers for things to work end-to-end.  For instance, I know that when I tried to read the actual TPM specs they were essentially incomprehensible.  We need to keep our summary of it to help developers using WebCrypto.  Likewise, the Android assertions will be commonly used, so it's a help to developers to keep a description of them.  We should take deleting this useful text off of the table.

Yes, we could move these descriptions into a different spec, but that only would make it harder for developers to understand end-to-end what they need to know to build systems using WebAuthN.  Therefore, I believe that we should leave things as they are.

**Comment by gmandyam** [comment]
But the spec does not provide sufficient information on the attestation formats.  For instance, the SafetyNet attestation links to a webpage hosted outside of the W3C, and in my opinion describes a proprietary format (e.g. there is no description how the boolean "ctsProfileMatch" is determined).  I don't see a purpose in providing descriptions of proprietary attestation formats in the specification - that can be done via a registry (pointing to an outside specification if necessary).  

Keeping the packed attestation in the spec is fine, as it is fully-defined in the WebAuthn specification.  The other stuff can go in a registry.

**Comment by selfissued** [comment]
The comment "The other stuff can go in a registry" reflects a misunderstanding of registries - per Jeff Hodges' comments on the list.  Normative content must be in a specification.  Registries simply contain references to those specifications.  So there's not an option to move normative content to a registry.  (We can move normative content to other specifications, should we choose to, but I think that would just make things harder for developers.)

Given that the SafetyNet attestation type will be widely used, if our description of it is presently inadequate, it would be my suggestion that we make it adequate, rather than deleting it.  Making things that WebAuthN developers will commonly use harder to find and use would do no favors to anyone.

**Comment by gmandyam** [comment]
No - my comment was "that can be done via a registry (pointing to an outside specification if necessary)."  The registry would have to point to normative content (i.e. "outside specification").

SafetyNet is already outdated; Android N attestation should be encouraged.  I don't think we favors for anyone pointing to attestation formats that are poorly defined and replaceable.

**Comment by equalsJeffH** [comment]
wrt  @gmandyam's comment https://github.com/w3c/webauthn/issues/108#issuecomment-235894994 ...
> @equalsjeffh wrote..
>> Re: "if we keep the specification of the presently-known webauthn attestation formats in the webauthn spec"

> In my opinion, the only attestation format specification format that might be kept in the spec is packed, because it is defined in the WebAuthn standard itself. Since the other types of attestations are defined by other organizations, 

I'm sorry, but I disagree with the latter claim -- other orgs may define the core components of the attestation facilities leveraged by webauthn, i.e., specifically that of TPMs and of Android SafetyNet -- we however need to specify **how** we leverage those facilities in the context of WebAuthn. It is not sufficient to simply point at the TPM specs and the SafetyNet specs and say "just do that stuff and plug the resultant blob in here".  

> the attestation registry can hold those references (to specifications for Android platform, TCG/TPM, etc).

As far as I understand, W3C specs can, and do, reference non-W3C specifications, and so having references to TPM and Android SafetyNet should not be an issue in and of themselves. Note also that they are not "Mandatory To Implement" in the context of the WebAuthn spec. 

> One possibility is to create a new W3C specification for packed attestation, and include the reference to it in the registry. This way, we can be internally consistent with the registry approach.

I do not agree that just because we (will) have a registry, that there is any valuable consistency to be gained by having all the attestation format webauthn-specific normative text in specs other then WebAuthn.  I think we ought to place in the WebAuthn spec the attestation format webauthn-specific normative text we know about at this time in the WebAuthn spec proper for essentially the same reasons @selfissued argues for doing so.

**Comment by gmandyam** [comment]
@equalsJeffH 
"As far as I understand, W3C specs can, and do, reference non-W3C specifications, and so having references to TPM and Android SafetyNet should not be an issue in and of themselves. "

I disagree with the premise that SafetyNet constitutes a "specification"  I think we are still ignoring that SafetyNet is not defined by any recognized SDO and lacks sufficient publicly-available information to be considered a standard.  A registry won't fix that, but I am not aware of any W3C specifications that reference proprietary formats.

In summary, I see nothing to be gained by referencing opaque, proprietary attestation formats that do not map to a standard.  That can go in a registry.

**Comment by equalsJeffH** [comment]
we've moved past this, closing.
